### PR TITLE
[9/x][programmable transactions] Added errors

### PR DESCRIPTION
**Pull request by tnowacki**
## Description 

Replaced panics with execution errors

## Test Plan 

👀 

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-adapter/src/programmable_transactions/context.rs [hunk 1]
```diff
@@ -186,47 +183,52 @@ where
         Ok(())
     }
 
-    /// Take the argument value, setting its value to None, making it unavailable
+    /// Get the argument value. Cloning the value if it is copyable, and setting its value to None
+    /// if it is not (making it unavailable).
     /// Errors if out of bounds, if the argument is borrowed, if it is unavailable (already taken),
     /// or if it is an object that cannot be taken by value (shared or immutable)
-    pub fn take_arg<V: TryFromValue>(
+    pub fn by_value_arg<V: TryFromValue>(
```
Comment (tnowacki) [inline crates/sui-adapter/src/programmable_transactions/context.rs:190]: Thanks for the suggestion @amnn!

File: crates/sui-types/src/messages.rs [hunk 2]
```diff
@@ -2357,6 +2376,22 @@ pub struct InvalidSharedByValue {
     pub object: ObjectID,
 }
 
+#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize, Hash)]
+pub enum CommandArgumentError {
```
Comment (amnn) [inline crates/sui-types/src/messages.rs:2380]: Would benefit from some docs:

- An overall doc comment reminding folks that these come from programmable transaction commands.
- A short description of how these errors come about (especially for the `Invalid*` errors -- what makes those things invalid?)

Comment (amnn) [reply crates/sui-types/src/messages.rs:2380]: (Or alternatively, a pointer in the outer doc comment to say that the `Display` implementation has this information 
 -- is there any chance we could use `thiserror` for this to bring these two things together?).

**Review by amnn — APPROVED** [review]
LGTM!

**Review by bmwill — APPROVED** [review]

File: crates/sui-types/src/messages.rs (hunk 2 above)
Comment (tnowacki) [reply crates/sui-types/src/messages.rs:2380]: Oooo fancy, will do

**Pull request merged**